radix-engine/src/vm/wasm_runtime/scrypto_runtime.rs

```diff
@@ -53,6 +53,34 @@ impl<'y, Y: SystemApi<RuntimeError>> ScryptoRuntime<'y, Y> {
             String::from_utf8(blueprint_name).map_err(|_| WasmRuntimeError::InvalidString)?;
         Ok((package_address, blueprint_name))
     }
+
+    #[cold]
+    pub fn consume_wasm_execution_units_unoptimized(
```

Comment:
MINOR comments:
* I don't think this needs to be `pub`
* I'm not particularly a fan of the `unoptimized` name, I think I'd slightly prefer the method name was a bit more semantic (e.g `consume_wasm_execution_exceeding_buffer`).
* If we have a semantic method name, we could simplify this method a little, and eek out a tiny bit more performance. I appreciate this is only hit about 100 times less frequently than the hot path, but I think it also helps with readability.
```rust
// We know this is > 0 based on the precondition of the method.
let n_remaining_after_buffer_used = n - self.wasm_execution_units_buffer;
let amount_to_request = n_remaining_after_buffer_used
    .checked_add(WASM_EXECUTION_COST_UNITS_BUFFER)
    .unwrap_or(u32::MAX);
self.api
    .consume_cost_units(ClientCostingEntry::RunWasmCode {
        package_address: &self.package_address,
        export_name: &self.export_name,
        wasm_execution_units: amount_to_request,
    })
    .map_err(InvokeError::downstream)?;
self.wasm_execution_units_buffer += amount_to_request - n_remaining_after_buffer_used;
```

Reply:
The last line doesn't seems correct. Should be `=` instead, right?

Reply:
Conceptually it could be `self.wasm_execution_units_buffer = WASM_EXECUTION_COST_UNITS_BUFFER` if we - but can't change that for now in case we ever exceed `u32::MAX` (somehow!) as we'd bill something different.

We've agreed to tweak this going forward when we tweak the instruction costing.